sample/DateTimeParser/DateTimeParser.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using Superpower;
+using Superpower.Parsers;
+
+namespace DateTimeParser
+{
+    static public class DateTimeParser 
+    {
+        private static TextParser<string> _2DigitParser =
+            from d1 in Character.Digit
+            from d2 in Character.Digit
+            select new string(new char[] {d1, d2});
+
+        private static TextParser<string> YearOfDate = 
+            from y1 in Character.Digit
+            from y2 in Character.Digit
+            from y3 in Character.Digit
+            from y4 in Character.Digit
+            select new string(new char[] {y1, y2, y3, y4});
+        private static TextParser<string> MonthOfDate = 
+            _2DigitParser;
+        private static TextParser<string> DayOfDate = 
+            _2DigitParser;
+
+        public static TextParser<DateTime> Date = 
+            from year in YearOfDate.Select(_ => Int32.Parse(_))
```

Comment:
Could this be simplified to:

```csharp
YearOfDate.Select(Int32.Parse)
```

Normally, `_` as an argument name means "unused".